### DISCUSSION TITLE
ELI5: The benefits of using "async" for this?

### DISCUSSION THREAD

**Discussion by perigvennetier** [boyan-soubachov/tastyworks_api#68] [Q&A] [answered]
It's all in the title, I'm just curious and learning. Thanks.

**Comment by bandwiches** [comment, marked as answer]
Use case would be real-time data sensitivity.  You don't want to slow down or pause real-time data for menial sub-tasks like retrieving watchlists and other things.  There is more to it than that, but essentially async programming allows you to off-load tasks into other threads so it doesn't block the main thread.
